olp-cpp-sdk-core/src/http/winhttp/NetworkWinHttp.cpp

```diff
@@ -0,0 +1,1006 @@
+    proxy_info.dwAccessType = WINHTTP_ACCESS_TYPE_NAMED_PROXY;
+    proxy_info.lpszProxy = const_cast<LPWSTR>(proxy_string.c_str());
+    proxy_info.lpszProxyBypass = WINHTTP_NO_PROXY_BYPASS;
+
+    DWORD error_code = ERROR_SUCCESS;
+    if (!WinHttpSetOption(http_request, WINHTTP_OPTION_PROXY, &proxy_info,
+                          sizeof(proxy_info))) {
+      error_code = GetLastError();
+      EDGE_SDK_LOG_WARNING(LOGTAG, "WinHttpSetOption(Proxy) failed "
+                                       << error_code << " "
+                                       << request.GetUrl());
+    }
+    if (!proxy.GetUsername().empty() && !proxy.GetUsername().empty()) {
+      std::wstring proxy_username;
+      const auto username_res =
+          ConvertMultiByteToWideChar(proxy.GetUsername(), proxy_username);
+
+      std::wstring proxy_password;
+      const auto password_res =
+          ConvertMultiByteToWideChar(proxy.GetPassword(), proxy_password);
+
+      if (username_res && password_res) {
+        LPCWSTR proxy_lpcwstr_username = proxy_username.c_str();
+        if (!WinHttpSetOption(http_request, WINHTTP_OPTION_PROXY_USERNAME,
+                              const_cast<LPWSTR>(proxy_lpcwstr_username),
+                              wcslen(proxy_lpcwstr_username))) {
+          error_code = GetLastError();
+          EDGE_SDK_LOG_WARNING(LOGTAG,
+                               "WinHttpSetOption(proxy username) failed "
+                                   << error_code << " " << request.GetUrl());
+        }
+
+        LPCWSTR proxy_lpcwstr_password = proxy_password.c_str();
+        if (!WinHttpSetOption(http_request, WINHTTP_OPTION_PROXY_PASSWORD,
+                              const_cast<LPWSTR>(proxy_lpcwstr_password),
+                              wcslen(proxy_lpcwstr_password))) {
+          error_code = GetLastError();
+          EDGE_SDK_LOG_WARNING(LOGTAG,
+                               "WinHttpSetOption(proxy password) failed "
+                                   << error_code << " " << request.GetUrl());
+        }
+      } else {
+        if (!username_res) {
+          error_code = GetLastError();
+          EDGE_SDK_LOG_WARNING(LOGTAG, "Proxy username conversion failure "
+                                           << error_code << " "
+                                           << request.GetUrl());
+        }
+        if (!password_res) {
+          error_code = GetLastError();
+          EDGE_SDK_LOG_WARNING(LOGTAG, "Proxy password conversion failure "
+                                           << error_code << " "
+                                           << request.GetUrl());
+        }
+      }
+    }
+  }
+
+  flags = WINHTTP_DECOMPRESSION_FLAG_ALL;
+  if (!WinHttpSetOption(http_request, WINHTTP_OPTION_DECOMPRESSION, &flags,
+                        sizeof(flags)))
+    handle->no_compression = true;
+
+  const auto& extraHeaders = request.GetHeaders();
+
+  std::wostringstream headerStrm;
```

Comment:
```suggestion
  std::wostringstream header_stream;
```
I wouldn't omit vowels here - refer to the C++ google style
https://google.github.io/styleguide/cppguide.html#General_Naming_Rules :

> int cstmr_id;              // Deletes internal letters.